OneBusAway/ui/stops/OBAStopViewController.m

```diff
@@ -363,9 +363,19 @@ - (void)deleteAlarmForArrivalAndDeparture:(OBAArrivalAndDepartureV2*)dep {
     NSURLRequest *request = [self.modelService.obaJsonDataSource requestWithURL:alarm.alarmURL HTTPMethod:@"DELETE"];
     [self.modelService.obaJsonDataSource performRequest:request completionBlock:^(id responseData, NSUInteger responseCode, NSError *error) {
         [self.modelDAO removeAlarmWithKey:dep.alarmKey];
+        // Replace with a new row and update it.
+        [self updateAlarmStateForRowAtIndexPath:[self indexPathForModel:dep]];
     }];
 }
 
+- (void)updateAlarmStateForRowAtIndexPath:(NSIndexPath *)indexPath {
+    OBADepartureRow *row = (OBADepartureRow *)[self rowAtIndexPath:indexPath];
+    OBADepartureRow *newRow = row;
```

Comment:
If you meant where I assigned `newRow = row`, it was because I thought I needed to replace the old one with a new row. 
However, I just updated the code so now it will only grab the current row which alarm is being added/removed and update it's alarm state. Then reload this row.